### PR TITLE
WooCommerce: Fix the weight and dimensions inputs by displaying the correct unit.

### PR DESCRIPTION
**Pull request by justinshreve**
This PR builds on #14827 and actually displays the correct units for weight and dimensions. I converted the various weight inputs (variations, global row, product) to use a new component instead that would load the correct unit, and made the existing dimensions do the same thing. I've defaulted to inches and pounds when nothing is loaded since our MVP is US stores.

To Test:
* Go to `http://calypso.localhost:3000/store/product/:site`
* Make sure the weight and dimensions inputs display correctly both in variations view and normal product view.
* Go to `wp-admin` and `Settings > Products` and change your weight and dimensions units.
* Refresh the product form and see the new units loaded correctly into the component.

See #14185.

### CODE REVIEW

File: client/extensions/woocommerce/app/products/product-form-simple-card.js [hunk 1]
```diff
@@ -60,28 +60,21 @@ const ProductFormSimpleCard = ( { product, editProduct, translate } ) => {
...
 				<FormFieldSet className="products__product-dimensions-input">
 					<FormLabel>{ translate( 'Dimensions' ) }</FormLabel>
 					<FormDimensionsInput
-						unit={ translate( 'in' ) }
 						dimensions={ product.dimensions }
 						onChange={ setDimension }
 					/>
 				</FormFieldSet>
 				<FormFieldSet className="products__product-weight-input">
 					<FormLabel>{ translate( 'Weight' ) }</FormLabel>
```
Comment (a8ci18n) [inline client/extensions/woocommerce/app/products/product-form-simple-card.js]: Hi! I've found a possible matching string that has already been translated **24** times:
`translate( 'Weight', { context: 'Dns Record'} )` *ES Score: 19*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/fix/woocommerce-units-display)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/components/form-dimensions-input/index.jsx [hunk 2]
```diff
@@ -1,56 +1,112 @@
...
+					name="width"
+					placeholder={ translate( 'W' ) }
+					type="number"
+					value={ dimensions && dimensions.width || '' }
+					onChange={ onChange }
+					className="form-dimensions-input__width"
+				/>
+				<FormTextInputWithAffixes
+					name="height"
+					placeholder={ translate( 'H' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/components/form-dimensions-input/index.jsx]: Hi! I've found a possible matching string that has already been translated **8** times:
`translate( 'h' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/components/form-dimensions-input/index.jsx [hunk 3]
```diff
@@ -1,56 +1,112 @@
...
+
+	render() {
+		const { className, noWrap, dimensions, onChange, translate, dimensionsUnit } = this.props;
+		const classes = classNames( 'form-dimensions-input', className, { 'no-wrap': noWrap } );
+
+		return (
+			<div className={ classes }>
+				<FormTextInput
+					name="length"
+					placeholder={ translate( 'L' ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/components/form-dimensions-input/index.jsx]: Hi! I've found a possible matching string that has already been translated **34** times:
`translate( 'L', { context: 'momentjs format string (sameElse)'} )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/app/products/product-form-variations-table.js [hunk 4]
```diff
@@ -8,12 +8,12 @@ import { find, isNumber } from 'lodash';
 /**
  * Internal dependencies
  */
+import CompactFormToggle from 'components/forms/form-toggle/compact';
```
Comment (allendav) [inline client/extensions/woocommerce/app/products/product-form-variations-table.js]: Why was this added?

File: client/extensions/woocommerce/components/form-dimensions-input/index.jsx (hunk 3 above)
Comment (allendav) [inline client/extensions/woocommerce/components/form-dimensions-input/index.jsx]: Please add a comment argument to give the translator context for this super short string :) https://github.com/Automattic/i18n-calypso#more-translate-examples

File: client/extensions/woocommerce/components/form-dimensions-input/index.jsx [hunk 5]
```diff
@@ -1,56 +1,112 @@
...
+					type="number"
+					value={ dimensions && dimensions.length || '' }
+					onChange={ onChange }
+					className="form-dimensions-input__length"
+				/>
+				<FormTextInput
+					name="width"
+					placeholder={ translate( 'W' ) }
```
Comment (allendav) [inline client/extensions/woocommerce/components/form-dimensions-input/index.jsx]: Same comment as above

File: client/extensions/woocommerce/components/form-dimensions-input/index.jsx (hunk 2 above)
Comment (allendav) [inline client/extensions/woocommerce/components/form-dimensions-input/index.jsx]: Same comment as above

File: client/extensions/woocommerce/components/form-dimensions-input/index.jsx [hunk 6]
```diff
@@ -1,56 +1,112 @@
...
+				/>
+			</div>
+		);
+	}
+}
+
+function mapStateToProps( state ) {
+	const site = getSelectedSiteWithFallback( state );
+	const dimensionsUnitSetting = site && getDimensionsUnitSetting( state, site.ID );
+	const dimensionsUnit = dimensionsUnitSetting && dimensionsUnitSetting.value || '';
```
Comment (allendav) [inline client/extensions/woocommerce/components/form-dimensions-input/index.jsx]: Maybe we could pick a default of 'in' instead?

File: client/extensions/woocommerce/components/form-weight-input/index.js [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+				size="4"
+			/>
+		);
+	}
+}
+
+function mapStateToProps( state ) {
+	const site = getSelectedSiteWithFallback( state );
+	const weightUnitSetting = site && getWeightUnitSetting( state, site.ID );
+	const weightUnit = weightUnitSetting && weightUnitSetting.value || '';
```
Comment (allendav) [inline client/extensions/woocommerce/components/form-weight-input/index.js]: Maybe fallback to 'lbs' instead

**Comment by allendav** [comment]
Note to testers: Requires installing the wc-api-dev plugin to work: https://github.com/woocommerce/wc-api-dev

File: client/extensions/woocommerce/app/products/product-form-variations-table.js (hunk 4 above)
Comment (justinshreve) [inline client/extensions/woocommerce/app/products/product-form-variations-table.js]: I moved the line from a few lines below because it was the only one not alphabetized.

File: client/extensions/woocommerce/components/form-dimensions-input/index.jsx [hunk 8]
```diff
@@ -1,56 +1,112 @@
...
+					name="width"
+					placeholder={ translate( 'W', { comment: 'Width placeholder for dimensions input' } ) }
+					type="number"
+					value={ dimensions && dimensions.width || '' }
+					onChange={ onChange }
+					className="form-dimensions-input__width"
+				/>
+				<FormTextInputWithAffixes
+					name="height"
+					placeholder={ translate( 'H', { comment: 'Height placeholder for dimensions input' } ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/components/form-dimensions-input/index.jsx]: Hi! I've found a possible matching string that has already been translated **8** times:
`translate( 'h' )` *ES Score: 9*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/extensions/woocommerce/components/form-dimensions-input/index.jsx [hunk 9]
```diff
@@ -1,56 +1,112 @@
...
+
+	render() {
+		const { className, noWrap, dimensions, onChange, translate, dimensionsUnit } = this.props;
+		const classes = classNames( 'form-dimensions-input', className, { 'no-wrap': noWrap } );
+
+		return (
+			<div className={ classes }>
+				<FormTextInput
+					name="length"
+					placeholder={ translate( 'L', { comment: 'Length placeholder for dimensions input' } ) }
```
Comment (a8ci18n) [inline client/extensions/woocommerce/components/form-dimensions-input/index.jsx]: Hi! I've found a possible matching string that has already been translated **34** times:
`translate( 'L', { context: 'momentjs format string (sameElse)'} )` *ES Score: 8*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).
